src/ida.rs

```diff
@@ -33,28 +33,15 @@ impl RabinIDA {
 
 impl RabinIDA {
     pub fn share(&self, data: Vec<u8>) -> Vec<RabinShare> {
-        let length = data.len();
         (1..=self.n)
-            .map(|x| {
-                let gx = GF(x);
-                RabinShare {
-                    id: x,
-                    length,
-                    body: data
-                        .chunks(self.k as usize)
-                        .map(|chunk| {
-                            chunk
-                                .into_iter()
-                                .rev()
-                                .fold(GF::zero(), |res, b| GF(*b) + gx * res)
-                                .into()
-                        })
-                        .collect(),
-                }
-            })
+            .map(|x| self.share_at_index(&data, x))
             .collect()
     }
 
+    pub fn share_at_index(&self, data: &Vec<u8>, index: u8) -> RabinShare {
+        calculate_share(&data, self.k, index)
```

Comment:
Any reason not to define `calculate_share(..)` inline here?

Reply:
Good point! I used that separate function in both methods first, but then re-used the method call. Might as well inline indeed!